docs/design/coreclr/botr/mscorlib.md

```diff
@@ -1,355 +1,356 @@
+    {
+        if (str == null)
+        {
+            throw new ArgumentNullException(nameof(str));
+        }
 
-Notice how oldBuffer and newBuffer (interior pointers into String instances) are re-fetched after allocating memory. Also, this method is an instance method in managed code, with the "this" parameter passed as the first argument. We use StringObject* as the argument type, then copy it into a STRINGREF so we get some error checking when we use it.
+        return str.IsInterned();
+    }
+}
+```
 
-	FCIMPL3(LPVOID, COMString::Replace, StringObject* thisRefUNSAFE, CLR_CHAR oldChar, CLR_CHAR newChar)
-	{
-	    FCALL_CONTRACT;
+### FCall example – unmanaged
 
-	    int length = 0;
-	    int firstFoundIndex = -1;
-	    WCHAR *oldBuffer = NULL;
-	    WCHAR *newBuffer;
+The FCall entrypoint has to be registered in tables in [vm\ecalllist.h][ecalllist] using `FCFuncEntry` macro. See ["Registering your QCall or FCall Method"](#register).
 
-	    STRINGREF   newString   = NULL;
-	    STRINGREF   thisRef     = (STRINGREF)thisRefUNSAFE;
+This method is an instance method in managed code, with the "this" parameter passed as the first argument. We use `StringObject*` as the argument type, then copy it into a `STRINGREF` so we get some error checking when we use it.
 
-	    if (thisRef==NULL) {
-	        FCThrowRes(kNullReferenceException, L"NullReference_This");
-	    }
+```C++
+FCIMPL1(Object*, AppDomainNative::IsStringInterned, StringObject* pStringUNSAFE)
+{
+    FCALL_CONTRACT;
 
-	    [... Removed some uninteresting code here for illustrative purposes...]
+    STRINGREF       refString   = ObjectToSTRINGREF(pStringUNSAFE);
+    STRINGREF*      prefRetVal  = NULL;
 
-	    HELPER_METHOD_FRAME_BEGIN_RET_ATTRIB_2(Frame::FRAME_ATTR_RETURNOBJ, newString, thisRef);
+    HELPER_METHOD_FRAME_BEGIN_RET_1(refString);
 
-	    //Get the length and allocate a new String
-	    //We will definitely do an allocation here.
-	    newString = NewString(length);
+    if (refString == NULL)
+        COMPlusThrow(kArgumentNullException, W("ArgumentNull_String"));
 
-	    //After allocation, thisRef may have moved
-	    oldBuffer = thisRef->GetBuffer();
+    prefRetVal = GetAppDomain()->IsStringInterned(&refString);
 
-	    //Get the buffers in both of the Strings.
-	    newBuffer = newString->GetBuffer();
+    HELPER_METHOD_FRAME_END();
 
-	    //Copy the characters, doing the replacement as we go.
-	    for (int i=0; i<firstFoundIndex; i++) {
-	        newBuffer[i]=oldBuffer[i];
-	    }
-	    for (int i=firstFoundIndex; i<length; i++) {
-	        newBuffer[i]=(oldBuffer[i]==((WCHAR)oldChar))?
-	                      ((WCHAR)newChar):oldBuffer[i];
-	    }
+    if (prefRetVal == NULL)
+        return NULL;
 
-	    HELPER_METHOD_FRAME_END();
+    return OBJECTREFToObject(*prefRetVal);
+}
+FCIMPLEND
+```
 
-	    return OBJECTREFToObject(newString);
-	}
-	FCIMPLEND
+<a name="register"></a>
 
+## Registering your QCall or FCall method
```

Comment:
```suggestion
## <a name="register" /> Registering your QCall or FCall method
```

I believe that this is the usual style used for this. Otherwise, it looks like part of previous paragraph when reading the raw text.